index.js

```diff
@@ -202,8 +202,18 @@
       var name = 'fantasy-land/' + _name;
       return requirements[_name] === Constructor ?
         function(typeRep) {
-          return funcPath([name], typeRep) ||
-                 implPath([/function (\w*)/.exec(typeRep)[1], name]);
+          var f = funcPath([name], typeRep);
+          if (f == null && typeof typeRep === 'function') {
+            /* istanbul ignore next */
+            var functionName = 'name' in function() {} ?
+                function functionName(f) { return f.name; } :
+                function functionName(f) {
+                  var match = /function (\w*)/.exec(f);
+                  return match && match[1];
+                };
```

Comment:
Why do you prefer `Function -> String`, @gabejohnson? Are you assuming that all environments which support arrow functions also support `Function#name`?